src/ring.fs

```diff
@@ -3,15 +3,15 @@ open System
 
 [<Struct>]
 type internal RingState<'item> =
-    | Writable of wx:'item array * ix:int
-    | ReadWritable of rw:'item array * wix:int * rix:int
+    | Writable of wx:'item option array * ix:int
```

Comment:
I understand the impulse to optionize, but the array is internal impl detail and there's nothing wrong with it using unchecked default for values because index range is tightly controlled anyway. From performance perspective having an option instead of value doubles the number of allocations and GC pressure.

Reply:
Yes this is a tricky thing with the Dart implementation right now. Dart has null safety so if I try to assign null to a generic it will complain and there's no such a thing as `default(T)` so I haven't found a way to allocate an array without providing a fill value.

At the moment, I'm compiling options as nullables in Dart so by using `'item option` I can assign null as default value. But I understand this has implications for .NET (would OptionValue help?) so I'm not asking to change it. Let's see if I manage to solve the issue in Dart, if not maybe we can also use `#if FABLE_COMPILER` here, although I'm also trying to avoid proliferation of conditional compilation :)

Reply:
yes, `voption` should work and I think would be more or less free from perf/cost perspective.